translator/jsonconfig/mergeJsonConfig.go

```diff
@@ -16,20 +17,29 @@ import (
 	"github.com/aws/amazon-cloudwatch-agent/translator/util"
 )
 
-func MergeJsonConfigMaps(jsonConfigMapMap map[string]map[string]interface{}, defaultJsonConfigMap map[string]interface{}, multiConfig string) (map[string]interface{}, error) {
+func MergeJsonConfigMaps(jsonConfigMapMap map[string]map[string]interface{}, defaultJsonConfigMap map[string]interface{}, multiConfig string, strictValidation bool, inputJsonDirPathErr error ) (map[string]interface{}, error) {
 	if jsonConfigMapMap == nil || len(jsonConfigMapMap) == 0 {
 		if os.Getenv(config.USE_DEFAULT_CONFIG) == config.USE_DEFAULT_CONFIG_TRUE {
 			// When USE_DEFAULT_CONFIG is true, ECS and EKS will be supposed to use different default config. EKS default config logic will be added when necessary
 			if ecsutil.GetECSUtilSingleton().IsECS() {
-				fmt.Println("No json config files found, use the default ecs config")
+				fmt.Println("Invalid agent's json config files, use the default ecs config")
 				return util.GetJsonMapFromJsonBytes([]byte(config.DefaultECSJsonConfig()))
 			}
 		}
-		if multiConfig == "remove" {
-			fmt.Println("No json config files found, please provide config, exit now")
-			os.Exit(config.ERR_CODE_NOJSONFILE)
+
+		if errors.Is(inputJsonDirPathErr, os.ErrNotExist) {
+			//When there is no input from flag --input, --input-dir and cannot find agent's config through containerized environment
+			log.Printf("No agent's json config was found from containerized environment or file path.")
+			if strictValidation{
+				os.Exit(config.ERR_CODE_NOJSONFILE)
+			}
+			log.Printf("Use the default agent's json config.")
+		} else if strictValidation || multiConfig == "remove" {
+			log.Printf("Invalid agent's json config files, please provide an appropriate config.")
+			os.Exit(config.ERR_CODE_INVALIDJSONFILE)
 		} else {
```

Comment:
I don't think we need this `else` here. Both of the conditions before it cause the process to end abruptly. If you took out the final else and kept the log statements, I feel like it would be just as valid.

Reply:
I would want to keep the ```log.Printf("Use the default agent's json config.")```. However, change code to node use the else after os.Exit 
```
		if isNotExistError {
			//When there is no input from flag --input, --input-dir and cannot find agent's config through containerized environment
			log.Printf("No agent's json config was found.")
			if strictValidation{
				os.Exit(config.ERR_CODE_NOJSONFILE)
			}
		} else {
			log.Printf("Invalid agent's json config files")
			if strictValidation || multiConfig == "remove" {
				os.Exit(config.ERR_CODE_INVALIDJSONFILE)
			}
		}
		
		log.Printf("Use the default agent's json config.")
		return defaultJsonConfigMap, nil
```